patches/server/0979-ExperienceOrb-should-call-EntitySpawnEvent.patch

```diff
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Golfing8 <atroo@me.com>
+Date: Mon, 8 May 2023 09:18:17 -0400
+Subject: [PATCH] ExperienceOrb should call EntitySpawnEvent
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index dee0168b50c7fbaefb762939a04e8f51e7bc58f7..666f1ebda945dc1709d086aa8925f950287b43de 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -689,7 +689,10 @@ public class CraftEventFactory {
+         } else if (entity instanceof net.minecraft.world.entity.ExperienceOrb) {
+             net.minecraft.world.entity.ExperienceOrb xp = (net.minecraft.world.entity.ExperienceOrb) entity;
+             double radius = world.spigotConfig.expMerge;
+-            if (radius > 0) {
++            // Paper start - Call EntitySpawnEvent for ExperienceOrb entities.
++            event = CraftEventFactory.callEntitySpawnEvent(entity);
++            if (radius > 0 && !event.isCancelled() && !entity.isRemoved()) {
++                // Paper end
```

Comment:
Yeah, I was unsure whether or not to merge the two Paper blocks or not since they have different purposes.

Reply:
Maybe something like this would be good? 
```diff
-             if (radius > 0) {
+            // Paper start - Call EntitySpawnEvent for ExperienceOrb entities.
+            event = CraftEventFactory.callEntitySpawnEvent(entity);
+            if (radius > 0 && !event.isCancelled() && !entity.isRemoved()) {
-             // Paper start - Maximum exp value when merging - Whole section has been tweaked, see comments for specifics
+            // Maximum exp value when merging - Whole section has been tweaked, see comments for specifics
              final int maxValue = world.paperConfig().entities.behavior.experienceMergeMaxValue;
              final boolean mergeUnconditionally = world.paperConfig().entities.behavior.experienceMergeMaxValue <= 0;
 ```
 In the end I would wait for a maintainer to tell you how you should do it. Interesting PR. Looking forward to it being merged!

Reply:
Leave it as a separate block. I've come to really hate combining paper comment blocks cause they you have to look at the blame to know what patch adds what cause you can have like 5 patches adding methods inside one block.